src/testnet/helium/operations.rs

```diff
@@ -0,0 +1,193 @@
+use std::io;
+use std::io::prelude::*;
+use std::io::{Read, Write};
+use rand::RngCore;
+
+use burnchains::{BurnchainSigner, PrivateKey};
+use chainstate::burn::{BlockHeaderHash, ConsensusHash, Opcodes, VRFSeed};
+use chainstate::stacks::StacksAddress;
+use net::StacksMessageCodec;
+use net::codec::{write_next};
+use net::Error as net_error;
+use util::hash::Hash160;
+use util::vrf::VRFPublicKey;
+use util::secp256k1::{MessageSignature, Secp256k1PublicKey, Secp256k1PrivateKey};
+
+
+#[derive(Debug, Clone)]
```

Comment:
It looks like you're duplicating all of the structs in `src/chainstate/burn/operations/`.  Why not just implement `StacksMessageCodec` for the existing operations?

Reply:
Yeah, using the structs from `src/chainstate/burn/operations/` force developer to work with bitcoin transaction objects, implying setting confusing fields such as txid, etc.
In this case, we just want to manipulate a blockstack op. I think that it should be the other way, and the structs in `src/chainstate/burn/operations/` should be splited and have an `op` field, being a struct like this ones. What do you think?

Reply:
I don't understand your concern.  If the "confusing fields" aren't even going to be used here, why not just implement a testnet-specific constructor for them and give them mocked values?  For example, the `txid` can just be a random string of bytes.  We need these fields to be present in the burn chain database either way, so we can't just remove them.  Splitting them out just makes it harder to load and store them in the burn chain database.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1381/commits/eb79e9ca7a2a4dcd40ca61f61adecf054a45e41a

Reply:
Thanks!